{"name": "psd-tools", "cve_id": "CVE-2020-10571", "publish_date": "2020-03-14T18:15Z", "cwe_id": "CWE-754", "owner": "Kota Yamaguchi", "affected_versions": [{"version_value": "1.9.4", "operator": "<"}], "references": ["https://github.com/psd-tools/psd-tools/security/advisories/GHSA-22jr-vc7j-g762", "https://github.com/advisories/GHSA-22jr-vc7j-g762", "https://github.com/psd-tools/psd-tools/releases/tag/v1.9.4", "https://github.com/psd-tools/psd-tools/pull/198", "https://nvd.nist.gov/vuln/detail/CVE-2020-10571"], "description": "From NVD:\nAn issue was discovered in psd-tools before 1.9.4. The Cython implementation of RLE decoding did not check for malicious data.\nFrom Github:\n### Impact\nAn issue was discovered in psd-tools before 1.9.4.\nThe Cython implementation of RLE decoding did not check for malformed PSD input data\nduring decoding to the PIL.Image or NumPy format, leading to a Buffer Overflow.\n\n### Patches\nUsers of psd-tools version v1.8.37 to v1.9.3 should upgrade to v1.9.4.\n\n### Workarounds\nWithout Cython present on installation, buffer overflow does not occur but IndexError will be thrown. However, already installed psd-tools with Cython extention should be upgraded.\n\n### References\nhttps://github.com/psd-tools/psd-tools/pull/198\n\n### For more information\nIf you have any questions or comments about this advisory:\n* Open an issue in [psd-tools](https://github.com/psd-tools/psd-tools/issues)", "latest_version": "1.9.16", "language": "python", "license": "MIT", "repository_summary": "Python package for working with Adobe Photoshop PSD files", "repo_url": "https://github.com/psd-tools/psd-tools", "impact": {"cvss2_score": 7.5, "cvss2_severity": "HIGH", "vector_string": "CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H", "av": "NETWORK", "ac": "LOW", "pr": "NONE", "c": "HIGH", "i": "HIGH", "a": "HIGH", "impact_score": 5.9, "exploitability_score": 3.9, "cvss3_score": 9.8, "cvss3_severity": "CRITICAL", "ui": "NONE", "s": "UNCHANGED"}, "ochrona_severity_score": 9.8}